modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/claim/metadata/mgt/ClaimUniquenessValidationTestCase.java

```diff
@@ -0,0 +1,516 @@
+
+        String user1Id = null;
+        String user2Id = null;
+        String user3Id = null;
+        try {
+            // Set claim uniqueness scope to WITHIN_USERSTORE
+            setClaimUniquenessScope(LocalClaimReq.UniquenessScopeEnum.WITHIN_USERSTORE);
+            Thread.sleep(2000);
+
+            // Create users with initial emails
+            UserObject user1 = initUser(PRIMARY_DOMAIN_ID, TEST_USER1_USERNAME, TEST_USER1_EMAIL);
+            CreateUserResponse response1 = scim2RestClient.attemptUserCreation(user1);
+            Assert.assertEquals(response1.getStatusCode(), HttpServletResponse.SC_CREATED);
+            user1Id = response1.getUserId();
+
+            UserObject user2 = initUser(PRIMARY_DOMAIN_ID, TEST_USER2_USERNAME, TEST_USER2_EMAIL);
+            CreateUserResponse response2 = scim2RestClient.attemptUserCreation(user2);
+            Assert.assertEquals(response2.getStatusCode(), HttpServletResponse.SC_CREATED);
+            user2Id = response2.getUserId();
+
+            UserObject user3 = initUser(SECONDARY_DOMAIN_ID, TEST_USER3_USERNAME, TEST_USER3_EMAIL);
+            CreateUserResponse response3 = scim2RestClient.attemptUserCreation(user3);
+            Assert.assertEquals(response3.getStatusCode(), HttpServletResponse.SC_CREATED);
+            user3Id = response3.getUserId();
+
+            PatchOperationRequestObject patchOp = createEmailUpdatePatchRequest(TEST_USER_EMAIL);
+
+            // User1 update should succeed
+            int updateStatus1 = scim2RestClient.attemptUserUpdate(patchOp, user1Id);
+            Assert.assertEquals(updateStatus1, HttpServletResponse.SC_OK, "User1 email update failed");
+
+            // User2 update should fail (same userstore as user1)
+            int updateStatus2 = scim2RestClient.attemptUserUpdate(patchOp, user2Id);
+            Assert.assertNotEquals(updateStatus2, HttpServletResponse.SC_OK, "User2 email update should have failed");
+
+            // User3 update should succeed (different userstore)
+            int updateStatus3 = scim2RestClient.attemptUserUpdate(patchOp, user3Id);
+            Assert.assertEquals(updateStatus3, HttpServletResponse.SC_OK, "User3 email update failed");
+
+        } finally {
+            // Clean up
+            if (user1Id != null) scim2RestClient.deleteUser(user1Id);
+            if (user2Id != null) scim2RestClient.deleteUser(user2Id);
+            if (user3Id != null) scim2RestClient.deleteUser(user3Id);
+        }
+    }
+
+    @Test(description = "Test email update with uniqueness validation across userstores (ACROSS_USERSTORES)")
+    public void testEmailUpdateAcrossUserStoresValidation() throws Exception {
+
+        String user1Id = null;
+        String user2Id = null;
+        String user3Id = null;
+        try {
+            // Set claim uniqueness scope to ACROSS_USERSTORES
+            setClaimUniquenessScope(LocalClaimReq.UniquenessScopeEnum.ACROSS_USERSTORES);
+            Thread.sleep(2000);
```

Comment:
Why do we need thread sleeps here?

Reply:
This is not required. Will remove it.

Reply:
Resolved via https://github.com/wso2/product-is/pull/22931/commits/96076a3195cd1279f7e0b36a14df2fa8dd620418